### DISCUSSION TITLE
.filter() is not working

### DISCUSSION THREAD

**Discussion by marcos-de-barros-vioque** [neuefische/questions#377] [Web] [answered]
Hi, 

I'm trying to filter and map the data from the form (which is firstly saved in `localStorage`) in the `CategoryOverview` component. This component is imported to the category pages (i.e. `food/index.js`).

The goal is to have the cards containing the form data displayed on the corresponding category page depending on the category selected by the visitor in the form (and without having the code for the filter/map methods on every category page).

This is what I've tried so far: 

```
<StyledLog>
          {spendingInput
            .filter(({category}) => category)
            .map(({index}) => {
              return (
                <StyledLogEntry key={index}>
                  <StyledLogCard>
                    {`${spendingInput.title} / ${spendingInput.amount} € / ${spendingInput.date} / ${spendingInput.category}`}
                  </StyledLogCard>
                </StyledLogEntry>
              );
            })}
        </StyledLog>
```

Link to the PR: https://github.com/marcos-de-barros-vioque/capstone/pull/22

I would appreciate your help. Thanks a lot in advance.

**Comment by badabam** [comment]
It seems you have solved this question. Please share the solution and a short description.

**Comment by marcos-de-barros-vioque** [comment, marked as answer]
Hi! In the meantime I could solve this topic {thanks, Stefan Korder!). 

What I had to was to define a new parameter in order to pass the data from category X to the corresponding category page (and include it in the .filter() with a strict equality operator):

**components/CategoryOverview.js:**

`export default function CategoryOverview({spendingInput, categoryName})`

`.filter(spendingInput => spendingInput.category === categoryName)`

In addition, I have added the parameter in the component import for the single pages, adapting the content of the string so that it matches the category value: 

**pages/food/index.js:** 

import CategoryOverview from "../../components/CategoryOverview";

```
const Food = ({spendingInput}) => (
  <CategoryOverview
    spendingInput={spendingInput}
    categoryName="Food"
  />
);

export default Food;
```
